daliuge-engine/dlg/deploy/slurm_client.py

```diff
@@ -260,15 +392,60 @@ def submit_job(self):
                 command = f"cd {session_dir} && sbatch --parsable {job_file_name}"
                 print(f"Submitting sbatch job: {command}")
                 stdout, stderr, exitStatus = remote.execRemote(
-                    self.host, command, username=self.username
+                    self.host, command, username=self.username, pkeyPath=self.ssh_key
                 )
                 if exitStatus != 0:
                     print(
-                        f"Job submission unsuccessful: {exitStatus.decode()}, {stderr.decode()}"
+                        f"Job submission unsuccessful: {exitStatus}, {stderr.decode()}"
                     )
                 else:
                     jobId = stdout.decode()
                     print(f"Job with ID {jobId} submitted successfully.")
         else:
             print(f"Created job submission script {job_file_name}")
+
+        return self.fetch_remote_status(jobId)
+    
+    def fetch_remote_status(self, jobId: str, timeout: int = 15):
```

Comment:
`timeout` is the wrong variable name here: it should be "`polling_gap`" or something. As it is currently implemented, this method doesn't timeout.